app/serializers/lists/protocol_serializer.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+module Lists
+  class ProtocolSerializer < ActiveModel::Serializer
+    include Canaid::Helpers::PermissionsHelper
+    include Rails.application.routes.url_helpers
+
+    attributes :name, :code, :keywords, :linked_tasks, :nr_of_versions, :assigned_users, :published_by,
+               :published_on, :updated_at, :archived_by, :archived_on, :urls, :default_public_user_role_id,
+               :hidden, :top_level_assignable, :has_draft, :team
+
+    def keywords
+      object.protocol_keywords.map(&:name)
+    end
+
+    def team
+      object.team.name
+    end
+
+    def linked_tasks
+      object.nr_of_linked_tasks
+    end
+
+    def assigned_users
+      object.user_assignments.map do |ua|
```

Comment:
Probably should use the `object.users` relation here (as you added it to the model anyway), it should do fewer DB queries.

Reply:
this action `user_name_with_role` must be against user_assignment to also fetch role. Logic same as before.